src/providers/ChatProvider.jsx

```diff
@@ -10,27 +10,35 @@ import { useFetch } from '@/hooks/useFetch';
 import URLs from '@/constants/constants';
 import { axiosClient } from '@/services/api';
 import { useSelector } from 'react-redux';
-import { getIsLoggedIn } from '@/redux-store/selectors';
+import { getIsLoggedIn, getToken } from '@/redux-store/selectors';
 
 const ChatContext = createContext();
 
 export const useChatContext = () => useContext(ChatContext);
 
 export const ChatProvider = ({ children }) => {
   const isUserLoggedIn = useSelector(getIsLoggedIn);
+  const token = useSelector(getToken);
   const [subscriptionRooms, setSubscriptionRooms] = useState([]);
   const [dataUserChats, setDataUserChats] = useState([]);
   const [filteredPrivateChats, setFilteredPrivateChats] =
     useState(dataUserChats);
   const [unreadRoomsCount, setUnreadRoomsCount] = useState(0);
   const [unreadDMsCount, setUnreadDMsCount] = useState(0);
   const [searchedValue, setSearchedValue] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (isUserLoggedIn && token) {
+      setIsLoading(false);
+    }
+  }, [isUserLoggedIn, token]);
 
   const { responseData: roomsData } = useFetch(
-    isUserLoggedIn ? URLs.userCountries : null
+    !isLoading && isUserLoggedIn && token ? URLs.userCountries : null
```

Comment:
Probably it can be good idea to move the condition out to the constant? `const checkLogin = !isLoading && isUserLoggedIn && token` and `checkLogin ? URLs.userCountries : null` something loke that. It`s just proposal.

Reply:
Good idea, done!